lib/CodeBuilder/Adapter/TolerantParser/Updater/ClassMethodUpdater.php

```diff
@@ -13,19 +14,20 @@
 use Phpactor\CodeBuilder\Domain\Prototype\Method;
 use RuntimeException;
 use Microsoft\PhpParser\Node;
+use Webmozart\Assert\Assert;
 
 /**
  * @extends AbstractMethodUpdater<ClassMembersNode|TraitMembers>
  */
 class ClassMethodUpdater extends AbstractMethodUpdater
 {
-    /**
-    * @return ClassMembersNode|TraitMembers|EnumMembers
-    */
-    public function memberDeclarationsNode(ClassLike $classNode)
+    public function memberDeclarationsNode(ClassLike|ObjectCreationExpression $classNode): ClassMembersNode|TraitMembers|EnumMembers
     {
-        if ($classNode instanceof ClassDeclaration) {
-            return $classNode->classMembers;
+        if ($classNode instanceof ClassDeclaration || $classNode instanceof ObjectCreationExpression) {
+            $classNode = $classNode->classMembers;
+            Assert::isInstanceOf( $classNode, ClassMembersNode::class);
```

Comment:
is it expected that this is NOT a ClassMembersNode? this will cause a runtime exception, which is probably not what we want.

if it should never happen then this should be an exception and I'd prefer not to use Assertion libraries.

Reply:
Yeah, the problem in this case is that it could also be a "normal" Object creation with a class name which would then crash here. But in this case I'm not sure what the best way to go about this is.